renku/domain_model/template.py

```diff
@@ -586,3 +588,73 @@ def update(self, template: Template):
         self.metadata["__template_id__"] = template.id
         self.metadata["__automated_update__"] = template.allow_update
         self.immutable_files = template.immutable_files
+
+
+def find_renku_section(lines: List[str]) -> Tuple[int, int]:
+    """Return start and end line numbers of the Renku-specific section."""
+    start = end = -1
+    for index, line in enumerate(lines):
+        if line.startswith("#        Renku-specific section - DO NOT MODIFY        #"):
```

Comment:
Because we often go back and forth on naming, configuration and similar things for templates can we make this a bit more general by looking in the template metadata for what is the start and end of this section and if there isn't anything in the template metadata then we default to what we have here.

This way if we change things in the future we can do so. Is that an easy thing to do? I think it would be nice if the template itself tells you what the "protected" start and end is.